src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,516 @@
+		'wp_ms_sites_list_table'                      => 'wp-admin/includes/class-wp-ms-sites-list-table.php',
+		'wp_ms_themes_list_table'                     => 'wp-admin/includes/class-wp-ms-themes-list-table.php',
+		'wp_ms_users_list_table'                      => 'wp-admin/includes/class-wp-ms-users-list-table.php',
+		'wp_plugin_install_list_table'                => 'wp-admin/includes/class-wp-plugin-install-list-table.php',
+		'wp_plugins_list_table'                       => 'wp-admin/includes/class-wp-plugins-list-table.php',
+		'wp_post_comments_list_table'                 => 'wp-admin/includes/class-wp-post-comments-list-table.php',
+		'wp_posts_list_table'                         => 'wp-admin/includes/class-wp-posts-list-table.php',
+		'wp_privacy_data_export_requests_list_table'  => 'wp-admin/includes/class-wp-privacy-data-export-requests-list-table.php',
+		'wp_privacy_data_removal_requests_list_table' => 'wp-admin/includes/class-wp-privacy-data-removal-requests-list-table.php',
+		'wp_privacy_policy_content'                   => 'wp-admin/includes/class-wp-privacy-policy-content.php',
+		'wp_privacy_requests_table'                   => 'wp-admin/includes/class-wp-privacy-requests-table.php',
+		'wp_screen'                                   => 'wp-admin/includes/class-wp-screen.php',
+		'wp_site_health_auto_updates'                 => 'wp-admin/includes/class-wp-site-health-auto-updates.php',
+		'wp_site_health'                              => 'wp-admin/includes/class-wp-site-health.php',
+		'wp_site_icon'                                => 'wp-admin/includes/class-wp-site-icon.php',
+		'wp_terms_list_table'                         => 'wp-admin/includes/class-wp-terms-list-table.php',
+		'wp_theme_install_list_table'                 => 'wp-admin/includes/class-wp-theme-install-list-table.php',
+		'wp_themes_list_table'                        => 'wp-admin/includes/class-wp-themes-list-table.php',
+		'wp_upgrader_skin'                            => 'wp-admin/includes/class-wp-upgrader-skin.php',
+		'wp_upgrader'                                 => 'wp-admin/includes/class-wp-upgrader.php',
+		'wp_users_list_table'                         => 'wp-admin/includes/class-wp-users-list-table.php',
+
+		/* Classes in wp-admin/includes/deprecated.php. */
+		'wp_user_search'                              => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_export_requests_table'       => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_removal_requests_table'      => 'wp-admin/includes/deprecated.php',
+	);
+
+	/**
+	 * Whether the autoloader has already been registered or not.
+	 *
+	 * Avoid registering the autoloader multiple times.
+	 *
+	 * @static
+	 * @access private
+	 *
+	 * @var bool
+	 */
+	private static $registered = false;
+
+	/**
+	 * Register the autoloader.
+	 *
+	 * @return void
+	 */
+	public static function register() {
```

Comment:
I haven't used `static::` myself, so just curious: Is there a difference to using `self::` here (as the class is `final` anyways)?

Reply:
It's my understanding that `self::` is a design fault in PHP, and you should rely on `static::` unless you need to request the current class's values explicitly. See "limitations of self::" at https://www.php.net/manual/en/language.oop5.late-static-bindings.php#language.oop5.late-static-bindings.self.

Reply:
These "limitations" are by design and have been how references work in PHP before late static binding was introduced.
For performance reasons, `self` should be preferred, as the class is indeed `final`. With a `final` class, the class cannot be extended, so you never need late static binding. Late static binding comes with extra overhead, as PHP will need to check whether the calling class has subclasses. With `self`, the code is simply executed without any additional checks.

So, generally, you'd go with either `final` + `private` + `self` OR `! final` + `protected` + `static`, unless there is a very specific reason to not do that because of the design.

Reply:
huh... I had no idea!! Thank you for the insights @schlessera!
Changed `static` to `self` in https://github.com/WordPress/wordpress-develop/pull/3470/commits/7188eb8827540858dc69a60da17584f3f06610a8 :+1: